packages/lesswrong/lib/routes.js

```diff
@@ -247,7 +247,7 @@ addRoute([
   },
   {
     name: 'comment.legacy',
-    path: `/${legacyRouteAcronym}/:id/:slug/:commentId`,
+    path: `/:section(r)?/:subreddit(all|discussion|lesswrong)?/${legacyRouteAcronym}/:id/:slug/:commentId`,
```

Comment:
I'm not quite sure about the syntax here – I'm guessing '?" means that the section of the url is optional. Is that correct?

Reply:
Correct, it means the segment (including the slash) is optional.